app/controllers/products_controller.rb

```diff
@@ -0,0 +1,50 @@
+class ProductsController < ApplicationController
+  def index
+    @title = "Products"
+    @vendor_id = params[:vendor_id]
+    @vendor = Vendor.find(@vendor_id)
```

Comment:
Since you're retrieving the `@vendor` information, you don't need a `@vendor_id` instance variable as well because you can use the `@vendor` variable to retrieve the id